### PR TITLE
Migration for benchmark tests

### PR DESCRIPTION
**Pull request by NeerajNagure**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
This PR migrates the shell benchmarking tests to e2e
#### Which issue(s) this PR fixes:

xref #1062 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/benchmark.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+				fmt.Printf("Error %s gap too large, actual: %d, expected: %d\n", resource, all-got, gap)
+				return fmt.Errorf("gap too large for resource %s", resource)
+			}
+		}
+		time.Sleep(1 * time.Second)
+	}
+}
+
+func scaleCreatePod(kwokctlPath string, name string, size int) error {
+	cmd := exec.Command(kwokctlPath, "--name", name, "kubectl", "get", "node", "-o", "jsonpath={.items.*.metadata.name}") // #nosec G204
```
Comment (wzshiming) [inline test/e2e/benchmark.go:67]:
```suggestion
func scaleCreatePod(ctx context.Context, kwokctlPath string, name string, size int) error {
	cmd := exec.CommandContext(ctx, kwokctlPath, "--name", name, "kubectl", "get", "node", "-o", "jsonpath={.items.*.metadata.name}") // #nosec G204
```

File: test/e2e/benchmark.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+	if err := scaleCmd.Start(); err != nil {
+		return fmt.Errorf("failed to start scale command: %w", err)
+	}
+	if err := waitResource(kwokctlPath, name, "Pod", "Running", size, 5, 1); err != nil {
+		return fmt.Errorf("failed to wait for resource: %w", err)
+	}
+	return nil
+}
+
+func scaleDeletePod(kwokctlPath string, name string, size int) error {
```
Comment (wzshiming) [inline test/e2e/benchmark.go:97]: ditto

File: test/e2e/benchmark.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+	}
+
+	if err := waitResource(kwokctlPath, name, "Pod", "fake-pod-", size, 0, 0); err != nil {
+		return fmt.Errorf("failed to wait for resource: %w", err)
+	}
+
+	return nil
+}
+
+func scaleCreateNode(kwokctlPath string, name string, size int) error {
```
Comment (wzshiming) [inline test/e2e/benchmark.go:113]: ditto

File: test/e2e/benchmark.go [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+			if len(failed) != 0 {
+				fmt.Println("------------------------------")
+				fmt.Println("Error: Some tests failed")
+				for _, test := range failed {
+					t.Fatalf(" - %s\n", test)
+				}
+			}
+			return ctx
+		})
+}
```
Comment (wzshiming) [inline test/e2e/benchmark.go:170]:
```suggestion
func CaseBenchmark(kwokctlPath, clusterName string) *features.FeatureBuilder {
	return features.New("Benchmarking").
		Assess("Create nodes", func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
			ctx0, _ := context.WithTimeout(ctx, 120*time.Second)
			err := scaleCreateNode(ctx0, kwokctlPath, clusterName, 2000)
			if err != nil {
				t.Fatal(err)
			}
			return ctx
		}).
		Assess("Create pods", func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
			ctx0, _ := context.WithTimeout(ctx, 240*time.Second)
			err := scaleCreatePod(ctx0, kwokctlPath, clusterName, 5000)
			if err != nil {
				t.Fatal(err)
			}
			return ctx
		}).
		Assess("Delete pods", func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
			ctx0, _ := context.WithTimeout(ctx, 240*time.Second)
			err := scaleDeletePod(ctx0, kwokctlPath, clusterName, 0)
			if err != nil {
				t.Fatal(err)
			}
			return ctx
		})
}
```

**Comment by NeerajNagure** [comment]
@wzshiming I have made all the changes

File: test/e2e/benchmark.go [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+		if got == want {
+			return nil
+		}
+		if got == prev {
+			fmt.Printf("Error %s %d not changed\n", resource, got)
+			return fmt.Errorf("resource %s not changed", resource)
+		}
+		prev = got
+		all := len(strings.Split(strings.TrimSpace(raw), "\n"))
+		fmt.Printf("%s %d/%d => %d\n", resource, got, all, want)
```
Comment (wzshiming) [inline test/e2e/benchmark.go:55]: I think we should use t(*testing.T).Log instead `fmt.Printf`

**Review by wzshiming — COMMENTED** [review]
Change the logging method and squash all commits.

**Comment by NeerajNagure** [comment]
@wzshiming I have replaced `fmt.Printf()` with `t.Logf()` and I have also squashed all the commits

**Pull request merged**